examples/generative/finetune_stable_diffusion.py

```diff
@@ -0,0 +1,507 @@
+* Shuffling and batching of the dataset.
+"""
+
+RESOLUTION = 256
+AUTO = tf.data.AUTOTUNE
+POS_IDS = tf.convert_to_tensor([list(range(MAX_PROMPT_LENGTH))], dtype=tf.int32)
+
+augmenter = keras_cv.layers.Augmenter(
+    layers=[
+        keras_cv.layers.CenterCrop(RESOLUTION, RESOLUTION),
+        keras_cv.layers.RandomFlip(),
+        tf.keras.layers.Rescaling(scale=1.0 / 127.5, offset=-1),
+    ]
+)
+text_encoder = TextEncoder(MAX_PROMPT_LENGTH)
+
+
+def process_image(image_path, tokenized_text):
+    image = tf.io.read_file(image_path)
+    image = tf.io.decode_png(image, 3)
+    image = tf.image.resize(image, (RESOLUTION, RESOLUTION))
+    return image, tokenized_text
+
+
+def apply_augmentation(image_batch, token_batch):
+    return augmenter(image_batch), token_batch
+
+
+def run_text_encoder(image_batch, token_batch):
+    return (
+        image_batch,
+        token_batch,
+        text_encoder([token_batch, POS_IDS], training=False),
+    )
+
+
+def prepare_dict(image_batch, token_batch, encoded_text_batch):
+    return {
+        "images": image_batch,
+        "tokens": token_batch,
+        "encoded_text": encoded_text_batch,
+    }
+
+
+def prepare_dataset(image_paths, tokenized_texts, batch_size=1):
+    dataset = tf.data.Dataset.from_tensor_slices((image_paths, tokenized_texts))
+    dataset = dataset.shuffle(batch_size * 10)
+    dataset = dataset.map(process_image, num_parallel_calls=AUTO).batch(batch_size)
+    dataset = dataset.map(apply_augmentation, num_parallel_calls=AUTO)
+    dataset = dataset.map(run_text_encoder, num_parallel_calls=AUTO)
+    dataset = dataset.map(prepare_dict, num_parallel_calls=AUTO)
+    return dataset.prefetch(AUTO)
+
+
+"""
+The baseline Stable Diffusion model was trained using images of 512x512 resolution. It's
+unlikely for a model that's trained using higher resolution images to transfer well to
+lower resolution images. However, the current model will lead to OOM if we keep the
+resolution to 512x512 (without enabling mixed-precision). Therefore, in the interest of
+interactive demonstration, we kept the input resolution to 256x256.
+"""
+
+# Prepare the dataset.
+training_dataset = prepare_dataset(
+    np.array(data_frame["image_path"]), tokenized_texts, batch_size=4
+)
+
+# Take a sample batch and investigate.
+sample_batch = next(iter(training_dataset))
+
+for k in sample_batch:
+    print(k, sample_batch[k].shape)
+
+"""
+We can also take a look at the training images and their corresponding captions.
+"""
+
+plt.figure(figsize=(20, 10))
+
+for i in range(3):
+    ax = plt.subplot(1, 4, i + 1)
+    plt.imshow((sample_batch["images"][i] + 1) / 2)
+
+    text = tokenizer.decode(sample_batch["tokens"][i].numpy().squeeze())
+    text = text.replace("<|startoftext|>", "")
+    text = text.replace("<|endoftext|>", "")
+    plt.title(text)
+
+    plt.axis("off")
+
+"""
+## A trainer class for the fine-tuning loop
+"""
+
+
+class Trainer(tf.keras.Model):
```

Comment:
same as above comment, we should consider providing a trainer in kerascv.
And thanks for the pointer here!